community/detectors/strapi_cve_2023_22893/src/test/java/com/google/tsunami/plugins/detectors/cves/cve202322893/Cve202322893VulnDetectorTest.java

```diff
@@ -0,0 +1,153 @@
+            .setBody(
+                "{\"jwt\":\"a jwt\n"
+                    + "token\",\"user\":{\"id\":2,\"username\":\"auth-bypass-example\",\"email\":\"notexists@notexist.com\",\"provider\":\"cognito\",\"confirmed\":true,\"blocked\":false,\"createdAt\":\"2023-04-28T06:56:20.344Z\",\"updatedAt\":\"2023-04-28T06:56:20.344Z\"}}")
+            .setResponseCode(DETECTION_STRING_BY_STATUS)
+            .setHeader(CONTENT_TYPE, MediaType.JSON_UTF_8.toString());
+    mockWebServer.enqueue(response);
+
+    DetectionReport actual =
+        detector.detect(targetInfo, ImmutableList.of(strapiService)).getDetectionReports(0);
+
+    DetectionReport expected =
+        DetectionReport.newBuilder()
+            .setTargetInfo(targetInfo)
+            .setNetworkService(strapiService)
+            .setDetectionTimestamp(Timestamps.fromMillis(Instant.now(fakeUtcClock).toEpochMilli()))
+            .setDetectionStatus(DetectionStatus.VULNERABILITY_VERIFIED)
+            .setVulnerability(
+                Vulnerability.newBuilder()
+                    .setMainId(
+                        VulnerabilityId.newBuilder()
+                            .setPublisher("TSUNAMI_COMMUNITY")
+                            .setValue("CVE_2023_22893"))
+                    .setSeverity(Severity.CRITICAL)
+                    .setTitle("Authentication Bypass for AWS Cognito Login Provider")
+                    .setDescription(
+                        "Strapi through 4.5.5 does not verify the access or ID tokens issued during the OAuth flow "
+                            + "when the AWS Cognito login provider is used for authentication.")
+                    .setRecommendation("Upgrade to higher versions")
+                    .addAdditionalDetails(
+                        AdditionalDetail.newBuilder()
+                            .setTextData(
+                                TextData.newBuilder()
+                                    .setText(
+                                        "A remote attacker could forge an ID token that is signed using the 'None' type algorithm "
+                                            + "to bypass authentication and impersonate any user that use AWS Cognito for authentication."
+                                            + " with the help of CVE-2023-22621 and CVE-2023-22894 attackers can gain "
+                                            + "Unauthenticated Remote Code Execution on this version of Strapi"))))
+            .build();
+    assertThat(actual).isEqualTo(expected);
+  }
+
+  @Test
+  public void detectWhenNotVulnerableReturnsNoVulnerability() {
+    mockWebServer.url("/notexistpath123321");
+    MockResponse response =
+        new MockResponse().setBody("NotExistDetectionString").setResponseCode(400);
+    mockWebServer.enqueue(response);
+
+    DetectionReportList findings = detector.detect(targetInfo, ImmutableList.of(strapiService));
+
+    assert (findings.getDetectionReportsList().isEmpty());
```

Comment:
Please use `assertThat` instead as Java assert statements are not run unless targets explicitly opt in via runtime flags to the JVM invocation.